build.sh

```diff
@@ -147,28 +164,77 @@ function get_and_build()
 	rm -rf "$tmp_path"
 }
 
-if [[ "$1" = "--list" ]]; then
-	echo "Fetching list from servers"
-	list_versions
-	exit
-elif [[ $# -lt 3 ]]; then
-	usage
-fi
+# Args handling
+opts="$(getopt -o 'h?' --long 'list,device:,os-version:,src:,dest-dir:' -- "$@" || { usage >&2 && exit 1; })"
+eval set -- "$opts"
+
+device=
+versions=
+module_dir=
+output_dir="output"
+
+while true; do
+    flag=$1
+    shift
+    case "$flag" in
+        --device) device="$1" && shift ;;
+		--os-version) versions="$1" && shift ;;
+		--dest-dir) output_dir="$1" && shift ;;
+        --src) module_dir="$1" && shift ;;
+        --) break ;;
+        *)
+            {
+                echo "error: unknown flag: $flag"
+                usage
+            } >&2
+            exit 1
+            ;;
```

Comment:
Guess it needs same spacing for all lines

Reply:
@floion, it looks fine to me, please correct me if i'm wrong.